packages/x-data-grid/src/components/GridHeaders.tsx

```diff
@@ -43,6 +44,7 @@ function GridHeaders() {
 
   const columnMenuState = useGridSelector(apiRef, gridColumnMenuSelector);
   const columnVisibility = useGridSelector(apiRef, gridColumnVisibilityModelSelector);
+  const columnGroupsModel = useGridSelector(apiRef, gridColumnGroupsUnwrappedModelSelector);
```

Comment:
Can we move this state extraction to the file where it's being used to avoid some props being drilled down?
```diff
diff --git a/packages/x-data-grid/src/hooks/features/columnHeaders/useGridColumnHeaders.tsx b/packages/x-data-grid/src/hooks/features/columnHeaders/useGridColumnHeaders.tsx
index 8d027322a..3266722a1 100644
--- a/packages/x-data-grid/src/hooks/features/columnHeaders/useGridColumnHeaders.tsx
+++ b/packages/x-data-grid/src/hooks/features/columnHeaders/useGridColumnHeaders.tsx
@@ -28,10 +28,8 @@ import {
   gridColumnPositionsSelector,
   gridVisiblePinnedColumnDefinitionsSelector,
 } from '../columns';
-import {
-  GridGroupingStructure,
-  GridColumnsGroupingState,
-} from '../columnGrouping/gridColumnGroupsInterfaces';
+import { GridGroupingStructure } from '../columnGrouping/gridColumnGroupsInterfaces';
+import { gridColumnGroupsUnwrappedModelSelector } from '../columnGrouping/gridColumnGroupsSelector';
 import { GridScrollbarFillerCell as ScrollbarFiller } from '../../../components/GridScrollbarFillerCell';
 import { getPinnedCellOffset } from '../../../internals/utils/getPinnedCellOffset';
 import { GridColumnHeaderSeparatorSides } from '../../../components/columnHeaders/GridColumnHeaderSeparator';
@@ -58,7 +56,6 @@ export interface UseGridColumnHeadersProps {
   headerGroupingMaxDepth: number;
   columnMenuState: GridColumnMenuState;
   columnVisibility: GridColumnVisibilityModel;
-  columnGroupsModel: GridColumnsGroupingState['unwrappedGroupingModel'];
   columnGroupsHeaderStructure: GridGroupingStructure[][];
   hasOtherElementInTabSequence: boolean;
 }
@@ -92,7 +89,6 @@ export const useGridColumnHeaders = (props: UseGridColumnHeadersProps) => {
     headerGroupingMaxDepth,
     columnMenuState,
     columnVisibility,
-    columnGroupsModel,
     columnGroupsHeaderStructure,
     hasOtherElementInTabSequence,
   } = props;
@@ -104,7 +100,7 @@ export const useGridColumnHeaders = (props: UseGridColumnHeadersProps) => {
   const theme = useTheme();
   const rootProps = useGridRootProps();
   const hasVirtualization = useGridSelector(apiRef, gridVirtualizationColumnEnabledSelector);
-
+  const columnGroupsModel = useGridSelector(apiRef, gridColumnGroupsUnwrappedModelSelector);
   const dimensions = useGridSelector(apiRef, gridDimensionsSelector);
   const columnPositions = useGridSelector(apiRef, gridColumnPositionsSelector);
   const renderContext = useGridSelector(apiRef, gridRenderContextColumnsSelector);
```